src/main/java/uk/gov/hmcts/reform/cpo/service/impl/CasePaymentOrdersServiceImpl.java

```diff
@@ -69,4 +85,40 @@ private void validateCasePaymentOrderQueryFilter(final CasePaymentOrderQueryFilt
                 "case-payment-orders cannot filter case payments orders by both id and cases-id.");
         }
     }
+
+    @Transactional
+    @Override
+    public CasePaymentOrder updateCasePaymentOrder(UpdateCasePaymentOrderRequest updateCasePaymentOrderRequest) {
+        String createdBy = securityUtils.getUserInfo().getUid();
+
+        // verify CPO exists upfront
+        CasePaymentOrderEntity casePaymentOrderEntity =
+            casePaymentOrdersRepository.findById(UUID.fromString(updateCasePaymentOrderRequest.getId()))
+                .orElseThrow(() -> new CasePaymentOrderCouldNotBeFoundException(ValidationError.CPO_NOT_FOUND));
```

Comment:
you could remove the comment and extract this block of code to a private method called
`verifyCPOexists`

Reply:
Resolved in 2052e030f2513753eabcf68894514c098487ae6a